src/upgrades/pre_0_14_3/mod.rs

```diff
@@ -0,0 +1,262 @@
+        }
+
+        Ok(true)
+    }
+}
+
+/// This function is used to recover from corrupt command files resulting
+/// from issue #1160 - if needed for Write Ahead Log (WAL) stores.
+///
+/// In case the store has no apparent issues in "warming" the entities, then
+/// no migration is started.
+///
+/// If any issue is found, then we replay all commands for each entity (aggregate)
+/// starting from the last snapshot, in as far as they can be parsed and applied,
+/// into new migrated aggregate store.
+pub fn upgrade_wal<W: WalSupport>(name: &Namespace, config: &Config) -> UpgradeResult<bool> {
+    let current_wal_store = WalStore::<W>::create(&config.storage_uri, name)?;
+    if current_wal_store.warm().is_ok() {
+        info!("No need to migrate store {name}");
+        Ok(false)
+    } else {
+        info!("Issue found in store {name}. Will attempt to fix it in upgrade.");
+
+        let cur_kv = KeyValueStore::create(&config.storage_uri, name)?;
+        let new_kv = KeyValueStore::create_upgrade_store(&config.storage_uri, name)?;
+
+        for handle in current_wal_store.list()? {
+            // Try to get an initialised wal support type. This can only error out in
+            // case there is a valid snapshot but it cannot be saved in the new store.
+            match migrate_wal_snapshot::<W>(&handle, &cur_kv, &new_kv)? {
+                None => continue, // Cannot migrate this one.
+                Some(mut wal) => {
+                    // Apply and migrate each next command. This will error out
+                    // only if we cannot store the next command in the new store.
+                    while migrate_next_wal_command(&handle, &mut wal, &cur_kv, &new_kv)? {
+                        debug!("migrated change set for {handle}, now at version: {}", wal.revision());
+                    }
+                    info!("Migrated {handle} in store {name}");
+                }
+            }
+        }
+
+        Ok(true)
+    }
+}
+
+/// Check the task store for corrupted tasks due to issue #1160, and if present
+/// drop them in place. This is safe to do because missing tasks are re-added at start up.
+pub fn upgrade_tasks(config: &Config) -> UpgradeResult<()> {
+    let task_store = kvx::KeyValueStore::new(&config.storage_uri, TASK_QUEUE_NS)
+        .map_err(|e| UpgradeError::Custom(format!("Cannot create task store: {}", e)))?;
+
+    let pending_scope = Scope::from_segment(segment!("pending"));
+    let running_scope = Scope::from_segment(segment!("running"));
+
+    for pending_key in task_store
+        .list_keys(&pending_scope)
+        .map_err(|e| UpgradeError::Custom(format!("Cannot read pending tasks: {}", e)))?
+    {
+        if task_store.get(&pending_key).is_err() {
+            warn!("Pending task could not be parsed. Dropping: {}", pending_key);
+            task_store.delete(&pending_key).map_err(|e| {
+                UpgradeError::Custom(format!("Cannot delete corrupt task: {}. Error: {}", pending_key, e))
```

Comment:
Rather than error out here, if this fails could you not try and wipe the whole store as is done in `upgrade_status()`?